language/testing-infra/module-generation/src/generator.rs

```diff
@@ -177,10 +177,29 @@ impl<'a> ModuleGenerator<'a> {
         }
     }
 
+    fn struct_type_parameters(&mut self) -> Vec<StructTypeParameter> {
+        // Don't generate type parameters if we're generating simple types only
+        if self.options.simple_types_only {
+            vec![]
+        } else {
+            let is_phantom = self.index(1) != 0;
+            let num_ty_params = self.index(self.options.max_ty_params);
+            let abilities = BTreeSet::from_iter(vec![Ability::Copy, Ability::Drop]);
+            init!(
+                num_ty_params,
+                (
+                    Spanned::unsafe_no_loc(TypeVar_::new(self.identifier())),
+                    abilities.clone(),
```

Comment:
same as above, I'd try to get all of the tuple args on one line, I find it a bit easier to read